### PR TITLE
feat: new tools, new node, upgrade

### PR DESCRIPTION
**Pull request by astagi**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+EXPOSE 3001
+EXPOSE 4000
+
+COPY ./Gemfile /app
+COPY ./Gemfile.lock /app
+COPY ./package-lock.json /app
+COPY ./package.json /app
+
+RUN bundle install
```
Comment (bfabio) [inline Dockerfile:13]:
```suggestion
RUN bundle config set --local deployment 'true'
RUN bundle install
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN npm i
```
Comment (bfabio) [inline Dockerfile:15]:
```suggestion
RUN npm ci
```

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+version: "3.9"
+services:
+  dev:
+    build: .
+    volumes:
+      - .:/app
+      - /app/node_modules
```
Comment (bfabio) [inline docker-compose.yml:7]:
```suggestion
      - /app/node_modules
      - /app/vendor
```

**Comment by astagi** [comment]
@bfabio thank you for your suggestions! implemented

**Pull request merged**